skyportal/models/observation_plan.py

```diff
@@ -0,0 +1,347 @@
+)
+ObservationPlanRequestTargetGroup.create = (
+    ObservationPlanRequestTargetGroup.update
+) = ObservationPlanRequestTargetGroup.delete = (
+    AccessibleIfUserMatches('followuprequest.requester')
+    & ObservationPlanRequestTargetGroup.read
+)
+
+
+class EventObservationPlan(Base):
+    """Tiling information, including the event time, localization ID, tile IDs,
+    and plan name"""
+
+    observation_plan_request_id = sa.Column(
+        sa.ForeignKey('observationplanrequests.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='ObservationPlanRequest ID',
+    )
+
+    observation_plan_request = relationship(
+        "ObservationPlanRequest",
+        foreign_keys=observation_plan_request_id,
+        doc="The request that this observation plan belongs to",
+    )
+
+    instrument_id = sa.Column(
+        sa.ForeignKey('instruments.id', ondelete="CASCADE"),
+        nullable=False,
+        doc='Instrument ID',
+    )
+
+    instrument = relationship(
+        "Instrument",
+        foreign_keys=instrument_id,
+        doc="The Instrument that this observation plan belongs to",
+    )
+
+    dateobs = sa.Column(
+        sa.ForeignKey('gcnevents.dateobs', ondelete="CASCADE"),
+        nullable=False,
+        index=True,
+        doc='UTC event timestamp',
+    )
+
+    plan_name = sa.Column(sa.String, unique=True, doc='Plan name')
+
+    validity_window_start = sa.Column(
+        sa.DateTime,
+        nullable=False,
+        default=datetime.utcnow,
+        doc='Start of validity window',
+    )
+
+    validity_window_end = sa.Column(
+        sa.DateTime,
+        nullable=False,
+        default=datetime.now() + timedelta(1),
+        doc='End of validity window',
+    )
+
+    status = sa.Column(
+        sa.String(),
+        nullable=False,
+        default="pending submission",
+        index=True,
+        doc="The status of the observing plan.",
+    )
+
+    planned_observations = relationship(
+        "PlannedObservation",
+        passive_deletes=True,
+        doc='Planned observations associated with this plan.',
+    )
+
+    @property
+    def start_observation(self):
+        """Time of the first planned observation."""
+        if self.planned_observations:
+            return min(
+                [
+                    planned_observation.obstime
+                    for planned_observation in self.planned_observations
+                ]
+            )
+        else:
+            return None
+
+    @hybrid_property
+    def num_observations(self):
+        """Number of planned observation."""
+        return len(self.planned_observations)
+
+    @num_observations.expression
+    def num_observations(cls):
+        """Number of planned observation."""
+        return cls.planned_observations.count()
+
+    @property
+    def total_time(self):
+        """Total observation time (seconds)."""
+        return sum(_.exposure_time for _ in self.planned_observations)
```

Comment:
Is there a reason to use underscores instead of a short variable name?

Reply:
fixed.